### DISCUSSION TITLE
EnterRaffle() function passing all test on Anvil but throwing an EVM revert error on Sepolia TestNet

### DISCUSSION THREAD

**Discussion by alfheimrShiven** [Cyfrin/foundry-full-course-cu#586] [Q&A] [answered]
I've deployed my `Raffle.sol` contract onto Sepolia TestNet and the contract seems to be verified as well. You can find the contract on Etherscan by visiting [https://sepolia.etherscan.io/address/0x8eb83d9ede9f044c3e004dcb71cd0df0b1413bcb](url)

I've been trying to interact with my contract from the Etherscan UI by calling the `enterRaffle()` function but it seems to be failing with the error

`Transaction has been reverted by the EVM: { "blockHash": "0x3cccfacbb6b4ce0d01ccbb3a9d3704a2923bf49134d08e0f9ce5bfdedb7c1c1d", "blockNumber": 4053896, "contractAddress": null, "cumulativeGasUsed": 1269936, "effectiveGasPrice": "0x65630044", "from": "0x689ecf264657302052c3dfbd631e4c20d3ed0bab", "gasUsed": 21339, "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "status": false, "to": "0x8eb83d9ede9f044c3e004dcb71cd0df0b1413bcb", "transactionHash": "0x49ea52981f1ae9aa2d97d7a03f89862e4a6b6bd43bd86d957786d567fd1a802c", "transactionIndex": 25, "type": "0x2", "events": {} }`

Attaching screenshot for your reference. 
<img width="1440" alt="Screenshot 2023-08-09 at 2 12 49 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/50408041/3579703d-285b-4d63-9c3a-c56f21db38b4">

Link to my repo: [https://github.com/alfheimrShiven/lottery-smart-contract](url)

Things I've checked:
1. Code passes all tests on Anvil.
2. My wallet has enough balance to execute the transaction.

Any observations will be helpful. Thanks!

**Comment by theirrationalone** [comment]
You forgot to import and inherit.... `AutomationCompatibleInterface`

```solidity
import {AutomationCompatibleInterface} from "@chainlink/contracts/src/v0.8/interfaces/AutomationCompatibleInterface.sol";

contract Raffle is VRFConsumerBaseV2, AutomationCompatibleInterface {
     // statements.
```

**Reply by alfheimrShiven** [reply]
Hey bro @theirrationalone thanks for pointing it out. I redeployed the contract after inheriting AutomationCompatibleInterface. I even registered and funded a new VRF upkeep, that would target the newly deployed contract for automation.
But unfortunately, the same error persists. 🤔

**Reply by theirrationalone** [reply]
Would you like to provide new deployment's contract address(sepolia/?)

**Reply by alfheimrShiven** [reply]
Yeah sure. `0x5C7Ba1b0C0460c511203C9f2b3D37D1A5134d07d`

**Reply by theirrationalone** [reply]
[Solution](https://github.com/Cyfrin/foundry-full-course-f23/discussions/586#discussioncomment-6681865) 

You're missing `getConversionRate(priceFeed)`

**Comment by theirrationalone** [comment]
You forgot to convert value == USD

that: `msg.value.getConversionRate(priceFeed)` ✅

At here:

```solidity
    function enterRaffle() public payable { 
         if (msg.value ❌ < i_entranceFee) { 
             revert Raffle__NotEnoughETHSent(); 
         } 
  
         if (s_raffleState != RaffleState.OPEN) { 
             revert Raffle__RaffleNotOpen(); 
         } 
  
         s_players.push(payable(msg.sender)); 
         emit EnteredRaffle(msg.sender); 
     }
```

**Reply by alfheimrShiven** [reply]
We are not using Chainlink's PriceFeed service in Raffle project.

**Reply by theirrationalone** [reply, marked as answer]
And you're sure that you're passing enogh EntranceFee

And one more thing if we're not using chainlink priceFeeds then Try to refactor that condition like this... We want strict entranceFee. 

```solidity
if (msg.value  != i_entranceFee) { 
             revert Raffle__NotEnoughETHSent(); 
      }
``

**Reply by theirrationalone** [reply]
I checked out your new deployment but everything looks good..

Possible issue you could have.... 

There can be some typos in your contract. 
Check your wallet balance, Maybe you're out of funds. 


You're cancelling `enterRaffle` transaction... I just pointed a performUpkeep transaction in your contract's transactions tab... This means you're intentionally doing this. 

this also means that your enterRaffle is working fine... 

Issue is with your subscriptions.

**Reply by alfheimrShiven** [reply]
Hey @theirrationalone , thanks for asking that question! I wasn't sending enough entrance fee! 
As you can see in my screenshot, I was sending 0.01 ether while the required is 0.1 ether. What a stupid mistake! But I wonder why the error msg on Etherscan does not show `Raffle__NotEnoughETHSent` revert msg. Instead it just shows `EVM Revert`🤦🏻‍♂️

**Reply by theirrationalone** [reply]
Maybe that error log was for miner on testnet though i'm not sure
